### ISSUE TITLE
Ability to include nested dotfiles

### ISSUE THREAD

**Issue by eknowlton** [audiolion/dotup#3]
I have a few files that I keep within `~/.config/` such as `~/.config/nvim` configuration as well as i3. 

It'd be very cool if it would create directories (if they don't exist) and then symlink those files within, recursively maybe?

**Comment by eknowlton** [comment]
One things I think of is that I have some folders that could be symlinked them selves but I also have some files that just sit next to each other and also sit next files generated by some applications I don't necessarily want to track.

Maybe I could do that with a `.gitignore` but for example...

I have a ...

```
.config/
.config/nvim
```

But sometimes VS Code puts a directory in config I don't want to track...
```
.config/
.config/nvim
.config/'Code - OSS <-- Don't want to track this folder.
```

**Comment by audiolion** [comment]
yeah this should probably be a thing. I will at the start just add recursive directory dotfile mapping, so if you have

```
dotfiles/
  | .bashrc
  | .config/
   -- nvim
```

if would create symlinks:

```
~/
  | .bashrc
  | .config/
   -- nvim
```

just mirroring the structure.

> But sometimes VS Code puts a directory in config I don't want to track...

This is probably related more to #5 , but since most people version their dotfiles as a git repo I could certainly ignore everything in the `.gitignore` folder, which might be a tidy way to have defaults, which you could override with `--include <file>` if you wanted to symlink something you were `.gitignore`-ing
